assets/src/js/admin/reports/components/tabs/index.js

```diff
@@ -8,11 +9,22 @@ import Tab from '../tab';
 const Tabs = () => {
 	const url = getWindowData( 'legacyReportsUrl' );
 
+	const renderItems = () => {
+		const items = applyFilters( 'givewp-reports-page-menu-items', [] );
+
+		if ( Array.isArray( items ) && items.length ) {
+			return items
+				.filter( ( item ) => item.hasOwnProperty( 'href' ) && item.hasOwnProperty( 'text' ) )
+				.map( ( item, i ) => <a key={ i } className="nav-tab" href={ item.href }>{ item.text }</a> );
+		}
+	};
```

Comment:
Couple suggestions:
- Make use of `useMemo` here to memoize the `items` variable, that way the items are only rendered when they change
- Change the hook to `givewp-reports-page-menu-links`, as I suspect we'll ultimately want to distinguish the links (i.e. anchors) from new pages, which will likely have its own register.
- Instead of filtering, do a check when mapping through the items and use `console.warn` to indicate improperly shaped items. Right now this would just silently ignore malformed items with no explanation. Guide devs in the right way to use something.

Reply:
@JasonTheAdams I did everything except making use of the useMemo function. The reason is very simple, `items` are not component props, so component won't be re-rendered each time React updates the DOM. If they were passed as component props, then we could use `React.memo` to prevent re-rendering, but in this case, it will be redundant.

Reply:
You are right, though I wish changes due to WordPress hooks did actually re-render the component. Oh well!